web/packages/teleport/src/Roles/RoleEditor/StandardEditor/standardmodel.ts

```diff
@@ -777,12 +782,14 @@ function roleConditionsToModel(
   const awsRoleARNsModel = aws_role_arns ?? [];
   const azureIdentitiesModel = azure_identities ?? [];
   const gcpServiceAccountsModel = gcp_service_accounts ?? [];
+  const mcpToolsModel = mcp?.tools ?? [];
```

Comment:
If the client was used to edit a role from the future version of the server, the `mcp` object might also contain unknown properties that would be silently obliterated. You should destructure it similarly to how it's done above and collect conversion errors from unknown properties using`unsupportedFieldErrorsFromObject`.

Reply:
something like this https://github.com/gravitational/teleport/pull/54734/commits/54c6c60f4c81f854b503dd75914232743f859938? still learning typescript